scripts/motor_driver.py

```diff
@@ -15,27 +15,43 @@ def __init__(self, pwm_pin: int, dir_pin: int, pi: pigpio.pi):
         self.pwm_pin = pwm_pin
         self.dir_pin = dir_pin
 
-    def set_speed(self, speed: float):
-        if speed < 0:
-            speed = -speed
-            dir_value = 1
-        else:
-            dir_value = 0
-
-        if speed > self.MAX_SPEED:
-            speed = self.MAX_SPEED
-
-        # Set direction
-        self.pi.write(self.dir_pin, dir_value)
-        
         # Set range to 0-100
         self.pi.set_PWM_range(self.pwm_pin, 100)
 
         # Set frequency to 8kHz
         self.pi.set_PWM_frequency(self.pwm_pin, 8000)
 
-        # Set duty cycle to speed
-        self.pi.set_PWM_dutycycle(self.pwm_pin, speed)
+        self.speed = 0
+
+    def set_speed(self, speed: float):
+        rate = rospy.Rate(204)  # 120hz
+
+        step = 2*self.MAX_SPEED/100 * (1 if speed > self.speed else -1)
+        instant_speed = self.speed
+        while abs(instant_speed - speed) > abs(step):
+            instant_speed += step
+
+            if instant_speed < 0:
+                abs_speed = -instant_speed
+                dir_value = 1
+            else:
+                abs_speed = instant_speed
+                dir_value = 0
+
+            if abs_speed > self.MAX_SPEED:
+                abs_speed = self.MAX_SPEED
+            
+            # Set direction
+            self.pi.write(self.dir_pin, dir_value)
+            # Set duty cycle to speed
+            self.pi.set_PWM_dutycycle(self.pwm_pin, abs_speed)
+
+            rospy.loginfo("Motor speed: " + str(instant_speed))
+
+            rate.sleep()
```

Comment:
I like the idea of a ramp. But I consider this is a separate feature that could go into its own function.

Reply:
@cychitivav 
Here is the branch 
https://github.com/cychitivav/pai/tree/fix/pwm-sign-jsduenass
Should I open a separate pull request?

Reply:
I think it is not necessary to create a new PR. In my opinion, it is better to solve #38 in this issue and I would propose that you suggest changes in this PR.

> PD: I have reviewed your branch and I like this approach, but I think it is still necessary to test this function in the lab.

Reply:
Thee problem is that you need to import np at the beginning so I find it easier to just add the commit instead of add a suggestion.